src/main/java/ch/hearc/todont/UserDetailsImplementation.java

```diff
@@ -0,0 +1,48 @@
+package ch.hearc.todont;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import ch.hearc.todont.models.ToDont;
+import ch.hearc.todont.models.User;
+import ch.hearc.todont.repositories.UserRepository;
+
+@Service
+public class UserDetailsImplementation implements UserDetailsService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Override
+    @Transactional(readOnly = true)
+	public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userRepository.findByName(username);
+        if (user == null) {
+            throw new UsernameNotFoundException(username);
+        }
+
+        // Should we keep this, and try to use it to manage access ?
+        Set<GrantedAuthority> grantedAuthorities = new HashSet<GrantedAuthority>();
```

Comment:
If it's not hard to check if user has the permissions `<id>_moderator`, we could probably keep that, but it kinda look finicky and is probably not the intended use of GrantedAuthority

Reply:
Sure, I wrote it _just in case_ it would prove useful later on. But it seems like it probably won't. Removing this on the next push.

Reply:
I meant the next next push